common/include/pcl/eigen_base.h

```diff
@@ -0,0 +1,150 @@
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ */
+
+#pragma once
+
+/**
+ * \file pcl/eigen_base.h
+ *
+ * \brief Defines Eigen related PCL functionality
+ * \ingroup common
+ */
+
+#include <cstdint>
+
+#include <Eigen/Core>  // for EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+
+/**
+ * \brief Macro to signal a class requires a custom allocator
+ *
+ *  It's an implementation detail to have pcl::has_custom_allocator work, a
+ *  thin wrapper over Eigen's own macro
+ *
+ * \see pcl::has_custom_allocator, pcl::make_shared
+ * \ingroup common
+ */
+#define PCL_MAKE_ALIGNED_OPERATOR_NEW \
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW \
+  using _custom_allocator_type_trait = void;
+
+#define PCL_ADD_EIGEN_MAPS_POINT4D \
+  inline pcl::Vector3fMap getVector3fMap () { return (pcl::Vector3fMap (data)); } \
+  inline pcl::Vector3fMapConst getVector3fMap () const { return (pcl::Vector3fMapConst (data)); } \
+  inline pcl::Vector4fMap getVector4fMap () { return (pcl::Vector4fMap (data)); } \
+  inline pcl::Vector4fMapConst getVector4fMap () const { return (pcl::Vector4fMapConst (data)); } \
+  inline pcl::Array3fMap getArray3fMap () { return (pcl::Array3fMap (data)); } \
+  inline pcl::Array3fMapConst getArray3fMap () const { return (pcl::Array3fMapConst (data)); } \
+  inline pcl::Array4fMap getArray4fMap () { return (pcl::Array4fMap (data)); } \
+  inline pcl::Array4fMapConst getArray4fMap () const { return (pcl::Array4fMapConst (data)); }
+
+#define PCL_ADD_POINT4D \
+  PCL_ADD_UNION_POINT4D \
+  PCL_ADD_EIGEN_MAPS_POINT4D
+
+#define PCL_ADD_UNION_NORMAL4D \
+  union EIGEN_ALIGN16 { \
+    float data_n[4]; \
+    float normal[3]; \
+    struct { \
+      float normal_x; \
+      float normal_y; \
+      float normal_z; \
+    }; \
+  };
+
+#define PCL_ADD_EIGEN_MAPS_NORMAL4D \
+  inline pcl::Vector3fMap getNormalVector3fMap () { return (pcl::Vector3fMap (data_n)); } \
+  inline pcl::Vector3fMapConst getNormalVector3fMap () const { return (pcl::Vector3fMapConst (data_n)); } \
+  inline pcl::Vector4fMap getNormalVector4fMap () { return (pcl::Vector4fMap (data_n)); } \
+  inline pcl::Vector4fMapConst getNormalVector4fMap () const { return (pcl::Vector4fMapConst (data_n)); }
+
+#define PCL_ADD_NORMAL4D \
+  PCL_ADD_UNION_NORMAL4D \
+  PCL_ADD_EIGEN_MAPS_NORMAL4D
+
+#define PCL_ADD_UNION_RGB \
```

Comment:
This macro has nothing to do with Eigen. I think only the macros that have `EIGEN_MAPS` in the name should be here.

Reply:
Ok, that was why my comment was asking for the line range.